crates/networking/p2p/src/network.rs

```diff
@@ -163,9 +171,25 @@ impl Network {
     }
 
     /// polling the libp2p swarm for network events.
-    pub async fn polling_events(&mut self) -> ReamNetworkEvent {
+    pub async fn polling_events(
+        &mut self,
+        mut request_rx: Receiver<NetworkRequest>,
+    ) -> ReamNetworkEvent {
         loop {
             tokio::select! {
+                 Some(request) = request_rx.recv() => {
+                match request {
+                    NetworkRequest::GetPeerCount(response_tx) => {
+                        let (connected, connecting, disconnecting, disconnected) = self.get_peer_counts();
```

Comment:
```suggestion
                        let (connected, connecting, disconnecting, disconnected) = self.get_peer_counts();
```

should just return the `PeerCountData` struct